src/v/cloud_storage/remote_partition.cc

```diff
@@ -1056,52 +928,90 @@ ss::future<> remote_partition::erase(ss::abort_source& as, bool do_finalize) {
         }
 
         vlog(
-          _ctxlog.debug,
-          "Erasing segments {}-{}",
+          cst_log.debug,
+          "[{}] Erasing segments {}-{}",
+          manifest.get_ntp(),
           *(batch_keys.begin()),
           *(--batch_keys.end()));
 
-        if (co_await tolerant_delete_objects(
-              _bucket, std::move(batch_keys), local_rtc)) {
-            co_return;
-        }
-
-        if (co_await tolerant_delete_objects(
-              _bucket, std::move(tx_batch_keys), local_rtc)) {
-            co_return;
-        }
-
-        if (co_await tolerant_delete_objects(
-              _bucket, std::move(index_keys), local_rtc)) {
-            co_return;
+        for (const auto& object_set : {batch_keys, tx_batch_keys, index_keys}) {
```

Comment:
nit: We could keep an in-memory hint of where we deleted successfully up to in order to skip some redundant requests. This is not a big deal for S3 or ABS as they support batch deletions, but may be worth it for GCP (can be done separately ofc)

Reply:
It's a bit awkward because the node doing this initial attempt at deletion is likely not to be the same one whose scrubber ends up doing the final cleanup.

I think to solve this in a robust way we'd need something like a deletion progress object in S3 (perhaps alongside the lifecycle marker that I'm adding in the next PR...) so that anyone doing the deletion from any node could save/restore their progress.

Unsure how much of an issue this will be in practice.  I guess the nasty case would be if we're getting lots of SlowDown responses, and we keep starting deletion, but never finishing.